manifests/plugin.pp

```diff
@@ -22,16 +22,24 @@
   $config_content  = undef,
   $update_url      = undef,
   $plugin_dir      = '/var/lib/jenkins/plugins',
-  $username        = 'jenkins',
-  $group           = 'jenkins',
   $enabled         = true,
   $create_user     = true,
   $source          = undef,
 ) {
   include ::jenkins::params
 
-  $plugin            = "${name}.hpi"
-  $plugin_parent_dir = inline_template('<%= @plugin_dir.split(\'/\')[0..-2].join(\'/\') %>')
+  $plugin     = "${name}.hpi"
+  $homedir    = $::jenkins::params::homedir
+  $jenkinsdir = $::jenkins::params::jenkinsdir
+  $username   = $::jenkins::params::jenkins_user
+  $group      = $::jenkins::params::jenkins_group
+
+  if $plugin_dir == '/var/lib/jenkins/plugins' {
+    $real_plugin_dir = $::jenkins::params::default_plugin_dir
+  } else {
+    $real_plugin_dir = $plugin_dir
+  }
+
```

Comment:
I'm not sure I follow the logic of what's happening here?

A selector would be more idiomatic way of assigning to $real_plugin_dir.